goethe/src/main/java/com/palantir/goethe/Goethe.java

```diff
@@ -172,6 +211,34 @@ public static Path formatAndEmit(com.squareup.javapoet.JavaFile file, Path baseD
         }
     }
 
+    /**
+     * Formats the given Java file and emits it to the appropriate directory under {@code baseDir}.
+     *
+     * @param fullClassName The class name of the java file, including package, eg {@code com.palantir.goethe.Goethe}
+     * @param source The java source code to format
+     * @param baseDir Source set root where the formatted file will be written
+     * @return the new file location
+     */
+    public static Path formatAndEmit(String fullClassName, String source, Path baseDir) {
+        String formatted = formatAsString(fullClassName, source);
+        try {
+            String packageName;
+            String className;
+            if (fullClassName.contains(".")) {
+                packageName = fullClassName.substring(0, fullClassName.lastIndexOf('.'));
+                className = fullClassName.substring(fullClassName.lastIndexOf('.') + 1);
+            } else {
+                packageName = "";
+                className = fullClassName;
+            }
+            Path output = getFilePath(baseDir, packageName, className);
```

Comment:
We're parsing this out just to put it back together again in `getFilePath`. It seems better to just add a `getFilePath` method that does what we need.

https://github.com/palantir/goethe/commit/2bc5d00a6d843dcf2446a94668f07e6b1320ab2c